internal/block/block_status.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package block
+
+// blockStatusInProgress
+type blockStatusInProgress struct {
+	state BlockState
+}
+
+func (b blockStatusInProgress) State() BlockState {
+	return b.state
+}
+
+func (b blockStatusInProgress) Error() error {
+	return nil
+}
+
+// blockStatusDownloadSuccess
+type blockStatusDownloadSuccess struct {
+	state BlockState
+}
+
+func (b blockStatusDownloadSuccess) State() BlockState {
+	return b.state
+}
+
+func (b blockStatusDownloadSuccess) Error() error {
+	return nil
+}
+
+// blockStatusDownloadError
+type blockStatusDownloadError struct {
+	state BlockState
+	err   error
+}
+
+func (b blockStatusDownloadError) State() BlockState {
+	return b.state
+}
+
+func (b blockStatusDownloadError) Error() error {
+	return b.err
+}
+
+// blockStatusDownloadCancelled
+type blockStatusDownloadCancelled struct {
+	state BlockState
+	err   error
+}
+
+func (b blockStatusDownloadCancelled) State() BlockState {
+	return b.state
+}
+
+func (b blockStatusDownloadCancelled) Error() error {
+	return b.err
+}
+
+// BlockStatus represents the status of a block download.
+// It provides methods to retrieve the current state of the block and any error
+// that may have occurred during the download process specifically for Failed state.
+type BlockStatus interface {
```

Comment:
why do you need an interface and multiple structs.
you can have either interface and 1 struct
or one public struct

Error will be nil incase of success.